internal/querynodev2/services.go

```diff
@@ -681,7 +681,13 @@ func (node *QueryNode) SearchSegments(ctx context.Context, req *querypb.SearchRe
 		return resp, nil
 	}
 
-	task := tasks.NewSearchTask(searchCtx, collection, node.manager, req, node.serverID)
+	var task tasks.Task
+	if paramtable.Get().QueryNodeCfg.UseStreamSearch.GetAsBool() {
+		task = tasks.NewStreamingSearchTask(searchCtx, collection, node.manager, req, node.serverID)
```

Comment:
Is it necessary to introduce a new type of task here? The future maintenance work could be expensive.

Reply:
The benefits for separating SearchTasks and StreamingSearchTasks：
1. logically safe and clear, we can totally disaggrate logics of batch reduce and stream reduce
2. As there're NQ-merge logic in querynode, the search tasks that will be using stream reduced should be isolated from search-tasks to be batch reduced.